AppCenterDataStorage/AppCenterDataStorage/MSDocumentWrapper.m

```diff
@@ -30,6 +33,37 @@ - (instancetype)initWithDeserializedValue:(id<MSSerializableDocument>)deserializ
   return self;
 }
 
+- (instancetype)initWithData:(NSData *)data documentType:(Class)documentType partition:(NSString *)partition {
+  if ((self = [super init])) {
+
+    // Try to deserialize the data.
+    NSError *deserializeError;
+    NSDictionary *json = [NSJSONSerialization JSONObjectWithData:data options:0 error:&deserializeError];
+
+    // Handle logging.
+    if (deserializeError) {
+      MSLogError([MSDataStore logTag], @"Error deserializing data: %@", [deserializeError description]);
```

Comment:
you should return nil here

Reply:
Well actually a document wrapper can store an error. It's probably what we should return there. If we were to list a partition and have some documents we cannot deserialize, the errors should surface this way.